system/src/system_control_internal.cpp

```diff
@@ -259,6 +344,150 @@ void particle::SystemControl::processRequest(ctrl_request* req, ControlRequestCh
         break;
     }
 #endif // Wiring_WiFi
+    case CTRL_REQUEST_SET_CLAIM_CODE: {
+        particle_ctrl_SetClaimCodeRequest pbReq = {};
+        int ret = decodeRequestMessage(req, particle_ctrl_SetClaimCodeRequest_fields, &pbReq);
+        if (ret == 0) {
+            ret = HAL_Set_Claim_Code(pbReq.code);
+        }
+        setResult(req, ret);
+        break;
+    }
+    case CTRL_REQUEST_IS_CLAIMED: {
+        const int ret = (HAL_IsDeviceClaimed(nullptr) ? SYSTEM_ERROR_NONE : SYSTEM_ERROR_NOT_FOUND);
```

Comment:
Is an error code the correct response for not claimed? I would imagine SYSTEM_ERROR_NONE to be returned in each case and the logical result returned as a response.

Reply:
I decided to save some space on simple getter requests returning `true` or `false`. Here is how the client side is implemented for this request: https://github.com/particle-iot/particle-usb/blob/master/src/cloud-device.js#L35

Let me know if you'd prefer this request to be reworked though.

Reply:
I'd prefer it to be reworked, since I feel it's semantically wrong to use an error code to indicate a negative result to a successful request. We can centralize the method for returning a boolean result, so the overhead is incurred only once.